### PR TITLE
Fix torch._numpy to match NumPy when empty ellipsis causes advanced indexing separation

### LINKED ISSUE #141563: Advanced indexing with empty ellipsis (discrepency to numpy)
### 🐛 Describe the bug

There is a difference in handling empty ellipses in advanced indexing.

The order of the result axes changes if the axes indexed are not next to each other. In that case, the new axes are prepended at the beginning. If they are neighboring, the are put in place of the original axes.

Here pytorch seems to check if an ellipsis actually covers any axes.
Numpy always treats it as covering axes and moves the new dimensions to the front

```python
import numpy as np
a=np.ones((3,4,5))
print(a[:,(0,1),...,(0,1)].shape)
# (2, 3)

import torch
a=torch.ones(3,4,5)
print(a[:,(0,1),...,(0,1)].shape)
# torch.Size([3, 2])
```

Is this an intentional deviation? Is this documented anywhere? Or is it a bug?

(If it is a bug, I am not sure if it is possible to fix this without breaking compilation or backward compatibility. )

### Versions

torch version: 2.5.1+cu121

### PR DESCRIPTION
**Pull request by manuelcandales**
Fixes #141563 

In NumPy, an ellipsis always acts as a separator between advanced indices, even when the ellipsis doesn't actually match any dimensions. In PyTorch an empty ellipsis doesn't cause a separation. This leads to differing behavior between Numpy and PyTorch in this edge case.

This difference in behavior leads to a bug when using torch.compile:
```python
>>> import numpy as np
>>> f = lambda x: x[:,(0,1),...,(0,1)].shape
>>> a = np.ones((3, 4, 5))
>>> f(a)
(2, 3)
>>> torch.compile(f)(a)
(3, 2)
```

Similarly to #157676, this PR doesn't change PyTorch's behavior, but it fixes the translation layer, ensuring torch._numpy compatibility with NumPy. I am marking this PR as fixing #141563, even though PyTorch behavior isn't modified.

Notice that there are still some other bugs in PyTorch's advanced indexing, that need to be fixed (mainly regarding proper accounting of dimensions when multidimensional boolean masks are present). But those need to be fixed at the ATen operator level. Examples:
- #71673
- #107699 
- #158125

cc @mruberry @rgommers

### CODE REVIEW

**Review by soumith — APPROVED** [review]
CI needs to be fixed but the patch looks good
